### DISCUSSION TITLE
The potential extension

### DISCUSSION THREAD

**Discussion by Maco1010** [Vinzent03/obsidian-git#827] [Q&A] [answered]
Hello !
Here is a question about some commands, such as I want to push something using the command push here.
![image](https://github.com/user-attachments/assets/00d02d86-4e3a-4ad5-aec6-a52f545f3f0e)
However, it seems that the 'push' command here actually encapsulates the git command 'git push' without the suffix options such as 'git push origin HEAD'. So could it be possible to develop some customized commands to make it more flexible to use git original command so that I don't need to switch to git bash to enter the git command again. 
Thank you!

**Comment by Vinzent03** [comment]
It should ask you for an upstream branch, which is always used for push. There's the "Set upstream branch" command to change which branch should be pushed to.
Does that suit your needs or what is the reason you always want push to a different remote?

**Reply by Maco1010** [reply]
Thanks for answering my question. My intention is that sometimes there is the need to use the customized command. For example I create a local branch named 'Develop',and I set upstream branch which accidentally named 'develop'. so far it is good. When I try to push with the button, it would remind that the name of the local branch differs from the remote branch. This time I need to switch to the git bash to push with the suffix option. Actually I mean that there maybe other cases the other options would be needed. But it is still ok with the present situation.

**Reply by Vinzent03** [reply]
But using the "Set upstream branch" command to fix your typo and use a different branch should work, doesn't it?

**Reply by Maco1010** [reply]
Well, I think I have to rename either branch and then it should work . But i don't mean to focus on this typo,  I mean would it be possible to allow the user to execute the git command with some options. That would be more perfect😀.

**Reply by Vinzent03** [reply, marked as answer]
Okay got it, I will think about providing a command to run arbitrary git commands.

**Reply by Vinzent03** [reply]
The "raw command" command is now available in v2.31.0 :tada:  which should fit your needs in this case.

**Reply by Maco1010** [reply]
Thanks! 😀😀😀🎉🎉🎉
